change to 3.19 EP2 waf configuration workflow (#1323)
tigera/docs: calico-enterprise/threat/web-application-firewall.mdx

  curl http://cartservice/cart?artist=0+div+1+union%23foo*%2F*bar%0D%0Aselect%23foo%0D%0A1%2C2%2Ccurrent_user
```

### Manage WAF configuration

> Reviewing the default rulset config:

Reviewer: `rule set` in all instances. I've put up a PR to address this elsewhere, but make sure any new copy in this PR follows this usage.

> ```conf

Reviewer: conf isn't a supported language for syntax highlighting. Let's leave this blank, or try ` ```bash `

Include @coraza.conf-recommended
Include @crs-setup.conf.example
Include @owasp_crs/*.conf

SecRuleEngine DetectionOnly
```

> The configuration file starts with importing the appropriate ruleset config. We use Coraza WAF's recommended [Core Ruleset setup](https://coraza.io/docs/tutorials/coreruleset/) files:

Reviewer: config --> configuration
Core Ruleset --> Core Rule Set

> 1. Coraza recommended [config](https://github.com/corazawaf/coraza/blob/main/coraza.conf-recommended)

Reviewer: ```suggestion
1. Coraza recommended [configuration](https://github.com/corazawaf/coraza/blob/main/coraza.conf-recommended)
```

> 1. The rest of the [coreruleset](https://github.com/coreruleset/coreruleset) files, currently [v4.0.0-rc2](https://github.com/coreruleset/coreruleset/tree/v4.0.0-rc2)

Reviewer: ```suggestion
1. The rest of the [Core Rule Set](https://github.com/coreruleset/coreruleset) files, currently [v4.0.0-rc2](https://github.com/coreruleset/coreruleset/tree/v4.0.0-rc2)
```

> These files can be customized if desired. 

Reviewer: ```suggestion
You can customize these files. 
```

1. (Recommended) Add all your customizations directly under `tigera.conf`

```bash
kubectl edit cm -n tigera-operator modsecurity-ruleset
``` 

> 1. (Not recommended) Starting your configmap from scratch

Reviewer: As we discussed, unless there's good reason to include this, it's best not to document a procedure we don't recommend. If you must include it, and you don't recommend it, explain why you don't recommend it.

| On     | Denies HTTP traffic. {{prodname}}  will log the event in Security Events. | Yes          |
| Off    | Be cautious about using this option. Traffic is not denied, and there are no events. |No                                                                      | Yes         |

##### Other basic customizations

For basic customizations, it's best to add it after all the includes in `tigera.conf`. In fact, this is the reason why the `SecRuleEngine` directive and the rest of [our customizations](https://github.com/tigera/operator/blob/master/pkg/render/applicationlayer/embed/coreruleset/tigera.conf#L8-L17) are situated there.

> An example is adding a sampling mode. For that, the tigera.conf will look like this:

Reviewer: ```suggestion
An example is adding a sampling mode. For that, the `tigera.conf` will look like this:
```

```conf

> # CoreRuleset activation

Reviewer: ```suggestion
# Core Rule Set activation
```
Make this change only if this is your comment, and not a comment that appears in the actual code. If that form is baked in, we should leave it as it actually appears.

Include @coraza.conf-recommended
Include @crs-setup.conf.example
Include @owasp_crs/*.conf

SecRuleEngine DetectionOnly

> # --- all customisations appear below this line, unless they need a specific loading order like plugins ---

Reviewer: ```suggestion
# --- all customizations appear below this line, unless they need a specific loading order like plugins ---
```
American spelling.

# --- Add sampling mode
# Read about sampling mode here https://coreruleset.org/docs/concepts/sampling_mode/
SecAction "id:900400,\
  phase:1,\
  pass,\
  nolog,\
  setvar:tx.sampling_percentage=50"
```

Also you can disable certain rules here:

```conf
# --- disable 'Request content type is not allowed by policy'
SecRuleRemoveById 920420
```

Change anomaly scoring threshold:

```conf
  SecAction \
  "id:900110,\
  phase:1,\
  nolog,\
  pass,\
  t:none,\
  setvar:tx.inbound_anomaly_score_threshold=25,\
  setvar:tx.outbound_anomaly_score_threshold=20"
```

> Or even change rule action parameters, behaviour e.g.:

Reviewer: ```suggestion
Or even change rule action parameters or behavior. For example:
```

 ```bash
 # --- append to more allowed content types to request bodies
SecAction \
  "id:900220,\
  phase:1,\
  nolog,\
  pass,\
  t:none,\
  setvar:'tx.allowed_request_content_type=|application/x-www-form-urlencoded| |multipart/form-data| |multipart/related| |text/xml| |application/xml| |application/soap+xml| |application/json| |application/cloudevents+json| |application/cloudevents-batch+json| |application/grpc|'"
```

> ##### Using coreruleset plugins

Reviewer: ```suggestion
##### Using Core Rule Set plugins
```

Let's go with an example plugin: [Wordpress Rule Exclusions](https://github.com/coreruleset/wordpress-rule-exclusions-plugin/).

Plugin files are the following:

```
wordpress-rule-exclusions-before.conf
wordpress-rule-exclusions-config.conf
```

To include these files properly, structure your work directory like so:

```
tigera.conf
wordpress-rule-exclusions-before.conf
wordpress-rule-exclusions-config.conf
```

> and then tigera.conf contents should be:

Reviewer: ```suggestion
and then `tigera.conf` contents should be:
```
